packages/compass-data-modeling/src/provider/index.tsx

```diff
@@ -62,9 +68,15 @@ export const DataModelStorageServiceProvider: React.FunctionComponent<{
             items: [...prevState.items],
           };
         });
-        return storageRef.current.save(toSave);
+        return true;
       },
       async delete(id: MongoDBDataModelDescription['id']) {
+        const deleted = await storageRef.current.delete(id);
+        // If delete fails in storage, we don't want to update the state
+        // to avoid showing the item in the list.
```

Comment:
Same question, if we are not updating state on saves like that, then we need to consider how to notify user about the failure, otherwise it's confusing. I don't understand why we need this change here though

Reply:
Like in `save`, if any storage method fails to change the state (within storage), changing state here means that the two are not in sync and for that reason i am checking if storage changed successfully, then update the state. 

And yes, i agree that we should notify the user if storage op fails (maybe a toast?)